packages/@react-aria/color/src/gradients.ts

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2022 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+*/
+
+export const generateRGB_R = (orientation, dir: boolean, zValue: number) => {
+  let maskImage = `linear-gradient(to ${orientation[Number(!dir)]}, transparent, #000)`;
+  let result = {
+    colorAreaStyles: {
+      backgroundImage: `linear-gradient(to ${orientation[Number(dir)]},rgb(${zValue},0,0),rgb(${zValue},255,0))`
+    },
+    gradientStyles: {
+      backgroundImage: `linear-gradient(to ${orientation[Number(dir)]},rgb(${zValue},0,255),rgb(${zValue},255,255))`,
+      'WebkitMaskImage': maskImage,
+      maskImage
+    }
+  };
+  return result;
+};
+
+export const generateRGB_G = (orientation, dir: boolean, zValue: number) => {
+  let maskImage = `linear-gradient(to ${orientation[Number(!dir)]}, transparent, #000)`;
+  let result = {
+    colorAreaStyles: {
+      backgroundImage: `linear-gradient(to ${orientation[Number(dir)]},rgb(0,${zValue},0),rgb(255,${zValue},0))`
+    },
+    gradientStyles: {
+      backgroundImage: `linear-gradient(to ${orientation[Number(dir)]},rgb(0,${zValue},255),rgb(255,${zValue},255))`,
+      'WebkitMaskImage': maskImage,
+      maskImage
+    }
+  };
+  return result;
+};
+
+export const generateRGB_B = (orientation, dir: boolean, zValue: number) => {
+  let maskImage = `linear-gradient(to ${orientation[Number(!dir)]}, transparent, #000)`;
+  let result = {
+    colorAreaStyles: {
+      backgroundImage: `linear-gradient(to ${orientation[Number(dir)]},rgb(0,0,${zValue}),rgb(255,0,${zValue}))`
+    },
+    gradientStyles: {
+      backgroundImage: `linear-gradient(to ${orientation[Number(dir)]},rgb(0,255,${zValue}),rgb(255,255,${zValue}))`,
+      'WebkitMaskImage': maskImage,
+      maskImage
+    }
+  };
+  return result;
+};
+
+
+export const generateHSL_H = (orientation, dir: boolean, zValue: number) => {
+  let result = {
+    colorAreaStyles: {},
```

Comment:
do you need to define an empty `colorAreaStyles` or can you not include it?

Reply:
it doesn't really matter, i like the uniformity though making it required